services/order.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+from django.shortcuts import get_object_or_404
+
+from db.models import Order, MovieSession, Ticket
+
+
+User = get_user_model()
+
+
+def create_order(
+    tickets: list[dict[str, int]],
+    username: str,
+    date: Optional[datetime] = None
+) -> None:
+    user = get_object_or_404(User, username=username)
+
+    with transaction.atomic():
+        order = Order.objects.create(user=user, created_at=date)
+
+        if date:
+            Order.objects.filter(id=order.id).update(created_at=date)
```

Comment:
The `created_at` field is being updated after the order is created, which is unnecessary if the date is provided during creation. Ensure that the `created_at` field is set correctly during the initial creation to avoid redundant updates.